MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm

```diff
@@ -626,25 +636,18 @@
 }
 
 void MVKInlineUniformBlockDescriptor::write(MVKDescriptorSet* mvkDescSet,
-									   VkDescriptorType descriptorType,
-									   uint32_t srcIndex,
-									   size_t stride,
-									   const void* pData) {
+                                            VkDescriptorType descriptorType,
+                                            uint32_t dstIndex,
+                                            size_t stride,
+                                            const void* pData) {
 	switch (descriptorType) {
 		case VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT: {
-			const auto& srcInlineUniformBlock = get<VkWriteDescriptorSetInlineUniformBlockEXT>(pData, stride, srcIndex);
-			_dataSize = srcInlineUniformBlock.dataSize;
-
-			[_mtlBuffer release];
-			if (srcInlineUniformBlock.dataSize > 0) {
-				MTLResourceOptions mtlBuffOpts = MTLResourceStorageModeShared | MTLResourceCPUCacheModeDefaultCache;
-				_mtlBuffer = [mvkDescSet->getMTLDevice() newBufferWithBytes: srcInlineUniformBlock.pData
-																	 length: srcInlineUniformBlock.dataSize
-																	options:mtlBuffOpts];	// retained
-			} else {
-				_mtlBuffer = nil;
-			}
-
+            const auto& pInlineUniformBlock = *(VkWriteDescriptorSetInlineUniformBlockEXT*)pData;
+			uint8_t* pDstData = (uint8_t*)_mtlBuffer.contents;
+            uint8_t* pSrcData = (uint8_t*)pInlineUniformBlock.pData;
+            if (pSrcData && pDstData) {
+                memcpy(pDstData + dstIndex, pSrcData, pInlineUniformBlock.dataSize);
+            }
```

Comment:
Moving the creation of the `MTLBuffer` out of this function was a really good catch. See my comment above about whether we need it at all, given your change to how the bytes are transferred to the encoder.

Reply:
I had to move it out because pInlineUniformBlock.dataSize can be less than the length of the buffer (thus we don't know the total length here). This is used for partial updates (with an offset) into the buffer as far as I can tell.